docs/plugin-development/restrictions.md

```diff
@@ -82,6 +82,44 @@ start debugging performance issues is through the Plugin Statistics window,
 which can be found through Plugins > Open Plugin Stats in the dev menu
 (`/xldev`).
 
+## Can my plugin talk to a server I run?
+
+Plugins are permitted to communicate with special backend servers, though there
+are certain considerations and requirements that must be met:
+
+- Plugins should take care to send the minimum amount of data necessary to do
+  their job. Whenever feasible, plugins should hash information like Content IDs
+  or player names on the client side so that a server-side data breach does not
+  reveal information.
+- Plugins collecting information for telemetry or analytics purposes must capture
+  as little information as necessary, and must clearly explain to the user what is
+  being collected and why.
+  - Users should be required to opt in to telemetry information, but this may be
+    done as part of a "welcome to this plugin" experience or controlled by a
+    Dalamud setting.
```

Comment:
I don't think uploads should be opt-in
Most users don't care, so if they don't even bother to opt-out, why would they bother to opt-in 

Same principal as silent agreement, you got informed that information may be uploaded, you can opt-out here, done
Example for this practice are both SubmarineTracker and Tracky, they inform the user and give them some hours to opt-out before any upload starts

Reply:
These plugins either have large focus on data collection (RNG) or are explicitly designed for it, which is a different court compared to something like Simple Tweaks installation information. IMO whether opt-in is required should become a case by case basis.

Reply:
I clarified this point a bit - opt-in is intended for plugins collecting "non-essential" information; that is, any information that is not required for a plugin to do its job. This would include things like how many people are using what features, any sort of census information (e.g. how many Miqo'te are using this pluign), and similar.

Things that are "essential" for a plugin's function (that is, if you don't want to submit that data, why are you even using this plugin) don't need an opt-in as the opt in is the very act of installing the plugin - or, ideally, a consent screen forcing a decision.

The intent here isn't to block telemetry as much as it is to encourage devs to consider the data they're collecting, why they're collecting it, and what that data is going to be used for. Any data collected should serve a purpose of some sort, and that purpose should (overall) be in line with our project's values.

Reply:
See also https://github.com/goatcorp/Dalamud/pull/1547, which will make Dalamud responsible for Analytics ID generation in a compliant fashion.